js/app/modules/card/thumb.js

```diff
@@ -125,9 +140,19 @@ const Thumb = new Module.Class({
         });
         this.add(this._layout);
 
+
+        this.connect('style-set', this._update_custom_style.bind(this));
+
         Utils.set_hand_cursor_on_widget(this);
     },
 
+    _update_custom_style: function () {
+        let context = this.get_style_context();
+        let thumb_background_size = EosKnowledgePrivate.style_context_get_custom_string(
+            context, 'thumbnail-background-size');
+        this.set_thumbnail_frame_from_model(this._thumbnail_frame, thumb_background_size);
```

Comment:
In any case, this should not call `set_thumbnail_frame_from_model` if the background size property didn't change.

Calling it more than once is problematic in the current code, because it doesn't remove the previous widget from the frame, so any subsequent calls will fail because the frame already has a child.